File: organiflow_app/lib/features/tasks/data/models/task_model.dart
import '../../domain/entities/task_entity.dart';
import 'form_schema_model.dart';

class TaskModel extends TaskEntity {
  const TaskModel({
    required super.id,
    required super.executionId,
    required super.workflowId,
    required super.nodeId,
    required super.nodeName,
    required super.status,
    super.assignedRole,
    super.departmentId,
    super.formSchema,
    super.formData,
    super.dueAt,
    super.completedAt,
    super.createdAt,
  });

  factory TaskModel.fromJson(Map<String, dynamic> json) {
    return TaskModel(
      id: json['id'] as String,
      executionId: json['executionId'] as String? ?? '',
      workflowId: json['workflowId'] as String? ?? '',
      nodeId: json['nodeId'] as String? ?? '',
      nodeName: json['nodeName'] as String? ?? '',
      status: json['status'] as String? ?? '',
      assignedRole: json['assignedRole'] as String?,
      departmentId: json['departmentId'] as String?,
      formSchema: json['formSchema'] != null
          ? FormSchemaModel.fromJson(json['formSchema'] as Map<String, dynamic>)
          : null,
      formData: json['formData'] as Map<String, dynamic>?,
      dueAt: json['dueAt'] != null
          ? DateTime.tryParse(json['dueAt'] as String)
          : null,
      completedAt: json['completedAt'] != null
          ? DateTime.tryParse(json['completedAt'] as String)
          : null,
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
    );
  }
}

File: organiflow-frontend/src/app/features/tasks/models/task.model.ts
export type TaskStatus = 'PENDING' | 'IN_PROGRESS' | 'DONE' | 'SKIPPED' | 'ESCALATED';
export type FieldType = 'text' | 'number' | 'select' | 'multiselect' | 'date' | 'file' | 'boolean' | 'textarea';

export interface FormField {
  name: string;
  label: string;
  type: FieldType;
  required: boolean;
  options: string[] | null;
  sortOrder: number | null;
}

export interface FormSchema {
  name: string;
  fields: FormField[];
}

export interface TaskResponse {
  id: string;
  tenantId: string;
  executionId: string;
  workflowId: string;
  nodeId: string;
  nodeName: string;
  assignedUserId: string | null;
  assignedRole: string | null;
  departmentId: string | null;
  status: TaskStatus;
  formSchema: FormSchema | null;
  formData: Record<string, unknown> | null;
  dueAt: string | null;
  completedAt: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface TaskCompleteRequest {
  formData: Record<string, unknown>;
}

export interface PreviousStepContext {
  nodeId: string;
  nodeName: string;
  completedAt: string | null;
  formData: Record<string, unknown> | null;
}

export interface AiFillFormRequest {
  transcript: string;
  formSchema: FormSchema;
  language?: string;
}

export interface AiFillFormResponse {
  fields: Record<string, unknown>;
  unfillableFields: string[];
}
